plugins/DeepJetTableProducer.cc

```diff
@@ -0,0 +1,315 @@
+          }
+          
+          // n_pf candidates
+          auto max_n_pf_n = std::min(features.n_pf_features.size(), (std::size_t)n_npf_);
+          for (std::size_t n_pf_n = 0; n_pf_n < max_n_pf_n; n_pf_n++) {
+            const auto& n_pf_features = features.n_pf_features.at(n_pf_n);
+            Npfcan_ptrel_nNpf[n_pf_n][i_jet] = n_pf_features.ptrel;
+            Npfcan_deltaR_nNpf[n_pf_n][i_jet] = n_pf_features.deltaR;
+            Npfcan_isGamma_nNpf[n_pf_n][i_jet] = n_pf_features.isGamma;
+            Npfcan_HadFrac_nNpf[n_pf_n][i_jet] = n_pf_features.hadFrac;
+            Npfcan_drminsv_nNpf[n_pf_n][i_jet] = n_pf_features.drminsv;
+            Npfcan_puppiw_nNpf[n_pf_n][i_jet] = n_pf_features.puppiw;
+          }
+
+          // sv candidates
+          auto max_sv_n = std::min(features.sv_features.size(), (std::size_t)n_sv_);
+          for (std::size_t sv_n = 0; sv_n < max_sv_n; sv_n++) {
+            const auto& sv_features = features.sv_features.at(sv_n);
+            sv_pt_nSV[sv_n][i_jet] = sv_features.pt;
+            sv_deltaR_nSV[sv_n][i_jet] = sv_features.deltaR;
+            sv_mass_nSV[sv_n][i_jet] = sv_features.mass;
+            sv_ntracks_nSV[sv_n][i_jet] = sv_features.ntracks;
+            sv_chi2_nSV[sv_n][i_jet] = sv_features.chi2;
+            sv_normchi2_nSV[sv_n][i_jet] = sv_features.normchi2;
+            sv_dxy_nSV[sv_n][i_jet] = sv_features.dxy;
+            sv_dxysig_nSV[sv_n][i_jet] = sv_features.dxysig;
+            sv_d3d_nSV[sv_n][i_jet] = sv_features.d3d;
+            sv_d3dsig_nSV[sv_n][i_jet] = sv_features.d3dsig;
+            sv_costhetasvpv_nSV[sv_n][i_jet] = sv_features.costhetasvpv;
+            sv_enratio_nSV[sv_n][i_jet] = sv_features.enratio;
+          }
+      }
+  }
+    
+  // DeepJetInputs table
+  auto djTable = std::make_unique<nanoaod::FlatTable>(jet_N_CPFCands.size(), nameDeepJet_, false, true);
+  //djTable->addColumn<int>("DeepJet_jetIdx", jetIdx_dj, "Index of the parent jet", nanoaod::FlatTable::IntColumn);
+    
+    
+  djTable->addColumn<int>("DeepJet_nCpfcand", jet_N_CPFCands, "Number of charged PF candidates in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_nNpfcand", jet_N_NPFCands, "Number of neutral PF candidates in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_nsv", jet_N_SVs, "Number of secondary vertices in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_npv", jet_N_PVs, "Number of primary vertices", nanoaod::FlatTable::IntColumn);
+    
+    
+  // ============================================================== Cpfs ===================================================================
+  for (unsigned int p = 0; p < n_cpf_; p++) {
+      auto s = std::to_string(p);
+      
+      djTable->addColumn<float>("DeepJet_Cpfcan_puppiw_" + s, Cpfcan_puppiw_nCpf[p], "charged candidate PUPPI weight of the " + s + ". cpf", nanoaod::FlatTable::FloatColumn, 10);
```

Comment:
```suggestion
      djTable->addColumn<float>("DeepJet_Cpfcan_puppiw_" + s, 
                                 Cpfcan_puppiw_nCpf[p], 
                                 "charged candidate PUPPI weight of the " + s + ". cpf", 
                                 nanoaod::FlatTable::FloatColumn, 10);
```

Reply:
can we format like this?

Reply:
yes we can! :) will change all affected lines and split them to improve readability.